api/wasm/v0/cpp/proxy_wasm_common.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Common enumerations available to WASM modules and shared with sandbox.
+ */
+// NOLINT(namespace-envoy)
+
+#pragma once
+
+#include <string>
+
+enum class WasmResult : uint32_t {
+  Ok = 0,
+  // The result could not be found, e.g. a provided key did not appear in a table.
+  NotFound = 1,
+  // An argument was bad, e.g. did not not conform to the required range.
+  BadArgument = 2,
+  // A protobuf could not be serialized.
+  SerializationFailure = 3,
+  // A protobuf could not be parsed.
+  ParseFailure = 4,
```

Comment:
ProtoParseFailure

Reply:
I was hoping to make these more general like the way errno error are more general (e.g. EIO EACCESS EINVAL) since they will always occur in the context of a call.   There are more things that can be serialized than protos.

Reply:
What does parsing or serialization mean outside of proto?

Reply:
There are other serialization systems than proto.  For example, one can serialize to json, flatbuffers, avro, thrift, cap'n proto, etc.